### ISSUE TITLE
Incompatible (or configuration is unclear) for fragments API and Typescript

### ISSUE THREAD

**Issue by sfarthin** [B2o5T/graphql-eslint#1427]
### Issue workflow progress

_Progress of the issue based on the
[Contributor Workflow](https://github.com/the-guild-org/Stack/blob/master/CONTRIBUTING.md#a-typical-contributor-workflow)_

- [x] 1. The issue provides a reproduction available on GitHub, Stackblitz or CodeSandbox

  https://codesandbox.io/p/sandbox/confident-kilby-wj0ssg

- [x] 2. A failing test has been provided
- [ ] 3. A local solution has been provided
- [ ] 4. A pull request is pending review

---

**Describe the bug**

Followed the prescribed setup for `@graphql-eslint/eslint-plugin` for inline GraphQL queries in TS files by inspecting the examples on https://github.com/B2o5T/graphql-eslint/tree/master/examples, but the rules do not seem to be applied.

**To Reproduce** Steps to reproduce the behavior:

1. Open   https://codesandbox.io/p/sandbox/confident-kilby-wj0ssg
2. Run `yarn eslint ./src --ext .tsx,.ts` in the terminal

**Expected behavior**

`yarn eslint ./src --ext .tsx,.ts` should trigger a `@graphql-eslint/no-anonymous-operations` error within `exampleQuery.ts`.

**Environment:**

- OS: codesandbox / macos
- `@graphql-eslint/eslint-plugin`: "3.10.6"
- Node.js: 16

**Additional context**

**Comment by B2o5T** [comment]
Run with `--ext` - `yarn eslint ./src`

**Comment by sfarthin** [comment]
For some reason, I am still not getting the lint error with `yarn eslint ./src`. Does this configuration looks OK?

```

**Comment by B2o5T** [comment]
Replace `files` for second override `*.{ts,tsx}` with `*.graphql`

**Comment by sfarthin** [comment]
I get this:

```
➜  workspace git:(master) ✗ yarn eslint ./src
yarn run v1.22.19
warning package.json: No license field
$ /project/home/sfarthing_rockfoc/workspace/node_modules/.bin/eslint ./src

/project/home/sfarthing_rockfoc/workspace/src/exampleQuery.tsx
  5:2  error  Parsing error: Unexpected keyword or identifier
```


This is a valid query and the codegen works as expected on it:

```ts
const Example = graphql(`
  query {
    foo
    bar
  }
`);
```


```js
module.exports = {
  parser: "@typescript-eslint/parser",
  overrides: [
    {
      files: ["**/*.tsx"],
      processor: "@graphql-eslint/graphql",
    },
    {
      files: ["*.graphql"],
      plugins: ["@graphql-eslint"],
      rules: {
        "@graphql-eslint/no-anonymous-operations": "error",
      },
    },
  ],
};
```

**Comment by B2o5T** [comment]
1. You don't run eslint for graphql files (even extracted from code-files)
2. You don't run graphql-eslint rules for `graphql` files
3. You parse graphql files with typescript-eslint parser

I suggest you to take a look of the working examples

**Comment by sfarthin** [comment]
> Move parser property to first override

no dice

```
➜  workspace git:(master) ✗ yarn eslint ./src
yarn run v1.22.19
warning package.json: No license field
$ /project/home/sfarthing_rockfoc/workspace/node_modules/.bin/eslint ./src

/project/home/sfarthing_rockfoc/workspace/src/exampleQuery.tsx
  5:9  error  Parsing error: Unexpected token {
```


> I suggest you to take a look of the working examples

I looked through them, but AFAICT theres no one example that matches

1. Uses TS 
2. Uses `ts` and `tsx` extensions
3. Inline GraphQL in code

I've tried lots of variations, but I get various parsing errors and its not clear to me what/when/where is be parsed wrong.

I might dig into the source some more to get a better understanding.

**Comment by sfarthin** [comment]
Why close this? I made an isolated example (https://codesandbox.io/p/sandbox/confident-kilby-wj0ssg) where you can replicate it easily.

**Comment by codepunkt** [comment]
@B2o5T I tried setting this up for 2 hours yesterday, without success.

Tried everything you said in this post & more, read all the documentation, looked at all the examples.
Nothing worked.

Even added `graphql-config` to the mix because I thought it might be related to that. It wasn't.

The example I'm looking for is `schema.graphql` outside of the package's root dir (monorepo), GraphQL queries exclusively in `ts` and `tsx` files using `graphql-tag`, no `/* GraphQL */` annotations on them.

**Comment by B2o5T** [comment]
@sfarthin 4. you miss `parser` to parse graphql files https://github.com/B2o5T/graphql-eslint/blob/80fbe116d9133bad06d7bb602df703cf2c2350dd/examples/svelte-code-file/.eslintrc.cjs#L19

@codepunkt provide a reproduction repo

**Comment by sfarthin** [comment]
> 4. you miss parser to parse graphql files

This was the last missing piece! I appreciate your work on this plugin and debugging this with me. Thank you!
